### PR TITLE
Audit log for admin actions

### PR DESCRIPTION
**Pull request by jrouzierinverse**
## Description

Audit admin action
## Impacts

The admin gui
## NEWS file entries

New Features
++++++++++++
- Admin audit log
## Delete branch after merge

NO

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Base/Controller/Crud.pm [hunk 1]
```diff
@@ -33,6 +33,21 @@ sub create :Local :Args(0) {
     }
 }
 
+sub audit_current_action {
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Base/Controller/Crud.pm]: As an old french adage goes :  'zi doc plize'

File: html/pfappserver/lib/pfappserver/Model/Audit.pm [hunk 2]
```diff
@@ -0,0 +1,94 @@
+package pfappserver::Model::Audit;
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Model/Audit.pm:1]: document the subroutines of this module

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm [hunk 3]
```diff
@@ -355,6 +360,24 @@ around create_action => sub {
     return $action;
 };
 
+=item audit_current_action
+
+Create an audit log entry
+
+=cut
+
+sub audit_current_action {
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm]: Is it really necessary to redefine it in this module ?

Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm]: Yes as this controller does not inherit from pfappserver::Base::Controller

However it does make more sense to create it as a role

Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Interface.pm]: Yes that sounds perfect

**Comment by jrouzierinverse** [comment]
Done please re-review

**Pull request merged**